scripts/canbench_ci_run_benchmark.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -Eexuo pipefail
+
+# Script that runs `canbench` at a given directory and outputs a comment
+# that is intended to be posted on the pull request.
+
+# Path to run `canbench` from.
+CANISTER_PATH=$1
+
+# Must match the file specified in the github action.
+COMMENT_MESSAGE_PATH=/tmp/canbench_comment_message.txt
+
+# Github CI is expected to have the main branch checked out in this folder.
+MAIN_BRANCH_DIR=_canbench_main_branch
+
+CANBENCH_OUTPUT=/tmp/canbench_output.txt
+
+CANBENCH_RESULTS_FILE="$CANISTER_PATH/canbench_results.yml"
+MAIN_BRANCH_RESULTS_FILE="$MAIN_BRANCH_DIR/$CANBENCH_RESULTS_FILE"
+
+# Install canbench
+cargo install canbench
+
+# Verify that canbench results are available.
+if [ ! -f "$CANBENCH_RESULTS_FILE" ]; then
+    echo "$CANBENCH_RESULTS_FILE not found. Did you forget to run \`canbench --persist\`?";
+    exit 1
+fi
+
+# Detect if canbench results file is up to date.
+pushd "$CANISTER_PATH"
+canbench --less-verbose > $CANBENCH_OUTPUT
```

Comment:
If I understand correctly, this one should run a benchmark against the master branch. Am I right? If so, please tell me when we checkout to the master branch?

Reply:
Actually I also got confused by the sequence. Seems the first call of `canbench` should be preceded with `pushd "$MAIN_BRANCH_DIR"`.

Reply:
We run canbench twice. The first time it's to check if `canbench_results.yml` in the current branch is up to date, that's this call. The second time happens on line 57 below, and that's where we compare with the master branch.

Reply:
ah, I see. I was misled by all the `pushd`/`pop` folder switching (which is a no-op in this case, since the canister dir is the same is root) -- I thought `canbench` is run in two branches.

Now I see that the first run is to check that the current branch has committed an up-to-date `canbench_results.yml` and then copy results from the main branch and run it again for comparison.